### PR TITLE
Add support for Platform 0.10

### LINKED ISSUE #1469: pack should support platform 0.10
Changes needed:

* Declare platform API 0.10 as supported
* Set `CNB_EXPERIMENTAL_MODE=warn` in the lifecycle's execution environment
* Invoke the restorer with the -build-image flag and cache volume mounted at /kaniko (when there are extensions)
* Invoke the extender binary with the builder image digest reference as the first argument and cache volume mounted at /kaniko (when there are extensions)
* Invoke the exporter as usual
* `pack inspect <image with 0.9 buildpacks>` should succeed (with existing codebase it shows `ERROR: unable to find image 'hello-extensions' locally or remotely`)

Out of scope (need separate issues):
* The `--extension` equivalent of `--buildpack`
* Support for extensions in project.toml
* `pack build` with extensions and `--cache-image` (adds complexity as it requires managing both a cache image and a cache volume; kaniko always needs a volume)
* `pack build` with extensions and a local (only) builder (because kaniko needs a manifest for the builder image, the builder must be pushed to a registry; would require a re-implementation of the extender in pack that uses the docker daemon instead of kaniko, or an `extender` lifecycle binary that can use the daemon socket)

### PR DESCRIPTION
**Pull request by natalieparellano**
## Summary
This PR builds off of https://github.com/buildpacks/pack/pull/1478 so will probably be easier to review when that is merged

Supersedes https://github.com/buildpacks/pack/pull/1527 (but keep that one open until we release `pack` with these changes as the docs PR points to https://github.com/buildpacks/pack/pull/1527)

Still a draft because:
* We need to update go.mod and the default lifecycle to 0.15.1 when released
* We need to add unit tests for `pack inspect` for platform API >= 0.10

## Documentation

- Should this change be documented?
    - [ ] Yes, see #___
    - [x] No

## Related

Resolves #1469

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -1,2 +1 @@
-* @buildpacks/platform-maintainers
-* @buildpacks/toc
+* @buildpacks/platform-maintainers @buildpacks/toc
```
Comment (natalieparellano) [inline CODEOWNERS:1]: 😢 I did something wrong

**Comment by natalieparellano** [comment]
This is ready for review but needs to be re-pointed when `extensions-phase-1` is merged as it is based off of that PR.

**Review by jjbustamante — APPROVED** [review]
LGTM ❤️

**Review by dfreilich — APPROVED** [review]
Looks really good! Overall, I think it could've used a bit more clarification in the code about why there were so many specific `kaniko` changes, but I assume that's covered in RFC.

**Pull request merged**